agents/labjack/labjack_agent.py

```diff
@@ -259,8 +283,32 @@ def start_acq(self, session, params=None):
         scan_rate_input = params.get('sampling_frequency',
                                      self.sampling_frequency)
         scans_per_read = max(1, int(scan_rate_input))
-        num_chs = len(self.chs)
-        ch_addrs = ljm.namesToAddresses(num_chs, self.chs)[0]
+
+        ### Filter stuff if specified
+        enable_filter = params.get('enable_filter', self.enable_filter)
+        lowpass_order = params.get('lowpass_order', self.lowpass_order)
+        lowpass_cutoff = params.get('lowpass_cutoff', self.lowpass_cutoff)
+        downsample_factor = params.get('downsample_factor',
+                                       self.downsample_factor)
+        save_unfiltered = params.get('save_unfiltered', False)
+        chs = params.get('channels', self.chs)
+        if not set(chs).issubset(set(self.chs)):
+            raise ValueError(
+                f"Invalid channels specified! Must be in {self.chs}"
+            )
+        self.log.info(f"Starting taking data for channels {chs}")
+        block_name = '_'.join(sorted(chs))
```

Comment:
Question about this: I was having trouble implementing the ability to set the channels being read out through the stream command because the block structure wouldn't match if you stopped the stream, and restarted the stream with a different channel set without restarting the agent. This is the solution I came up with (to make sure the block-name is unique for each combination of channels you could chose. But I just wanted to make sure this wouldn't break anything in the data-loading. I.e.  a field "AIN0V" happens to appear in two different blocks in a single frame, would the hk data loader know how to handle that? Should I change this so every channel is published to a unique block?

Reply:
I think that might violate something in the specification (like that a field must not appear in more than one block name for an active provider).  The intended solution for cases like this is that you resign the active provider and start a new provider.  The natural way to do that is to restart the agent.

Alternatively, you could publish to a different feed name (rather than same feed but different block name)... that would trigger a new provider instance (IIRC) **but** then the downstream systems wouldn't naturally group the fields together.

In summary... I don't think this is supported naturally.  A solution might be to add a trigger in the aggregator so it could somehow be told to re-register a provider, so you can change the field contents in a block.

Reply:
Ok I think I'll probably just remove this arg. Was useful for testing but not sure how useful it'll be in reality